Dockerfile

```diff
@@ -0,0 +1,33 @@
+FROM golang:alpine AS build_base
+ 
+RUN apk update --no-cache && apk add git
+ 
+WORKDIR /app
+ 
+ENV GO111MODULE=on
+ 
+COPY go.mod .
+ 
+COPY go.sum .
+ 
+RUN go mod download
+ 
+FROM build_base as build
+  
+COPY ./ /app
+ 
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -ldflags '-w -extldflags "-static"' -o proxy .
+ 
+FROM alpine
```

Comment:
I generally prefer alpine for the images I build, just to keep them smaller. There are cgo issues with alpine due to the different libc (which is why we had to use the debian version of the cloud build image to run the race detector), but since this is explicitly `CGO_ENABLED=0` and `-static`, alpine should be fine here (sans comment made in my upcoming full review).